src/Wallabag/CoreBundle/Command/UpdatePicturesPathCommand.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Wallabag\CoreBundle\Command;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Wallabag\CoreBundle\Repository\EntryRepository;
+
+class UpdatePicturesPathCommand extends Command
+{
+    private EntityManagerInterface $entityManager;
+    private EntryRepository $entryRepository;
+    private string $wallabagUrl;
+
+    public function __construct(EntityManagerInterface $entityManager, EntryRepository $entryRepository, $wallabagUrl)
+    {
+        $this->entityManager = $entityManager;
+        $this->entryRepository = $entryRepository;
+        $this->wallabagUrl = $wallabagUrl;
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('wallabag:update-pictures-path')
+            ->setDescription('Update the path of the pictures for each entry when you changed your wallabag instance URL.')
+            ->addArgument(
+                'old-url',
+                InputArgument::REQUIRED,
+                'URL to replace'
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $oldUrl = $input->getArgument('old-url');
+
+        $entries = $this->entryRepository->findAll();
```

Comment:
I think loading all entries is not very scalable, especially when there are lot of entries 😅 
maybe use `toIterable()`? https://www.doctrine-project.org/projects/doctrine-orm/en/2.15/reference/batch-processing.html

Reply:
@yguedidi I made a change about the findAll(). Not sure if it's ok ...

Reply:
To me it's still hydrating all entries in memory..
I think the exact fix is similar to https://www.doctrine-project.org/projects/doctrine-orm/en/2.15/reference/batch-processing.html#iterating-results
- create a select all QUery
- then use toIterate() in the loop

![image](https://github.com/wallabag/wallabag/assets/1480128/4de2d053-b367-4259-a25c-1d8d3eee0433)

Reply:
@yguedidi is that better?